app/Transcripts.tsx

```diff
@@ -0,0 +1,161 @@
+    const endDateString = filters.dateRange.end !== null ? formatDateForPath(filters.dateRange.end) : null;
+
+    getAllVideosForDateRange(filters.category, startDateString, endDateString)
+      .then(videoData => {
+        if (!ignore) {
+          setLoading(false);
+          setVideos(videoData);
+        }
+      });
+
+      return () => {
+        ignore = true;
+      }
+  }, [filters])
+
+  function handleFilterChange(filters: TranscriptFilterSelection) {
+    updateFilters(filters);
+    updateFilterParams(filters);
+  }
+
+  const videoLinks: React.ReactNode[] = videos.map(
+    video => (
+      <li key={video.videoId} className="mx-3 list-disc">
+        <Link href={getVideoPath(filters.category, video.videoId)}>
+          {video.title}
+        </Link>
+      </li>
+    ));
+
+  // Get access to the tailwind theme colors to pass to the loading spinner
+  // Specifying only content and theme keys is a workaround for a type error pertaining
+  // to the "future" key
+  const fullConfig = resolveConfig({ content, theme });
+  const loadingSection = <section className="my-4 flex flex-row">
+    <TailSpin wrapperClass="justify-center" color={fullConfig.theme.colors.blue['500']} />
+  </section>;
+
+  const resultsSection: React.ReactNode = <section>
+    <h2 className="my-4 text-lg">
+      Transcripts:
+    </h2>
+    <ul className="flex flex-col flex-wrap h-screen">
+      {videoLinks}
+    </ul>
+  </section>;
+
+  return (
+    <>
+      <TranscriptFilter selection={filters} onFilterChange={handleFilterChange} />
+      {isLoading ? loadingSection : resultsSection}
+    </>
+  );
+}
+
+function useFilterParams(allCategories: string[]): {
+   filterParams: TranscriptFilterSelection,
+   updateFilterParams: (newFilters: TranscriptFilterSelection) => void
+} {
+  const searchParams: URLSearchParams = useSearchParams();
+  const pathName: string = usePathname();
+  const { push } = useRouter();
+
+  const filterParams: TranscriptFilterSelection = {
+    category: getCategoryFromParams(searchParams, allCategories),
+    dateRange: getDateRangeFromParams(searchParams),
+  };
+
+  const updateFilterParams = (newFilters: TranscriptFilterSelection) => {
+    push(buildUrl(pathName, newFilters), { scroll: false });
+  }
+
+  return {
+    filterParams,
+    updateFilterParams
+  };
+}
+
+function getCategoryFromParams(params: URLSearchParams, allCategories: string[]): string {
+  const categoryParam: string | null = params.get('category');
+
+  if (categoryParam !== null && allCategories.includes(categoryParam)) {
+    return categoryParam;
+  }
+  else {
+    return 'sps-board';
+  }
+}
+
+function getDateRangeFromParams(params: URLSearchParams): DateRange {
+  const start: Date | null = getDateFromParams(params, 'start');
+  const end: Date | null = getDateFromParams(params, 'end');
+
+  if (start === null && end === null) {
+    return { start: parseDateFromPath('2024-03-01'), end: null };
```

Comment:
Pull default date into constant pelase

Reply:
Done, but the constant default is a hack IMO anyway. I want to think of a better way of coming up with a default start date. One thought is checking when the most recent transcript is from, and then subtracting a month from that.